bip-0325.mediawiki

```diff
@@ -26,10 +26,10 @@ A new type of test network would be more suitable for integration testing by org
 
 A new type of network ("signet"), which takes an additional consensus parameter called the challenge (scriptPubKey). The challenge can be a simple pubkey (P2PKH style), or a k-of-n multisig, or any other script you would want.
 
-The witness commitment of the coinbase transaction is extended to include a secondary commitment (the signature/solution) of either:
+Signet requires all blocks to have a BIP 141 commitment in the coinbase transaction. The BIP 141 commitment's optional data must include an additional commitment of the signature/solution for the block:
```

Comment:
This seems like it contradicts the text from #983: "There is one other acceptable special case: if a block's challenge is e.g. `OP_TRUE` (`0x51`), where an empty solution would result in success, the block is also considered valid if the signet commitment is absent"

Reply:
Yeah, good point. @instagibbs there is an exception now for the case where if the challenge is enough to satisfy (e.g. `OP_TRUE`); in this case, the signet commitment may be omitted entirely.

Reply:
In which case the BIP 141 commitment isn't required either

Reply:
How about:

> Signet requires all blocks to have a BIP 141 commitment in the coinbase transaction. **In order to provide a solution to the block challenge,** the BIP 141 commitment's optional data must include an additional commitment of the solution for the block: [...]. **In the special case where an empty solution is valid (ie scriptSig and scriptWitness are both empty) this additional commitment can optionally be left out. This special case is to allow non-signet-aware block generation code to be used to test a custom signet chain where the challenge is `OP_TRUE`.**

Reply:
I'd probably say "where the challenge is satisfied with no scriptsig or witness data" since I believe that's the more general case since things like CLEANSTACK aren't enforced?

Reply:
Took a modified version of what you posted, noting that bip141 commitment is only required with non-empty solutions, and that any "trivial" challenge can be used to avoid requiring fancy mining software for non-signature tests.

Reply:
I believe the current code requires a bip141 commitment even with empty solutions. Should this be changed?

Reply:
you are correct: https://github.com/bitcoin/bitcoin/pull/18267/commits/d0f364c2ddf6b7e7a1d89af18c60de8248ce5ff3#diff-4bdd7c44636dae0429a4d61909eeed17R90

I think for simplicity it's fine to require it. Any objections?

Reply:
Simplicity is good. Let's require